File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/QuizAttemptResponse.java
package com.hoangphihiep.dto.response;

import lombok.*;
import java.util.Date;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuizAttemptResponse {
    private Integer id;
    private String userId;
    private Integer quizId;
    private Double score;
    private Double totalScore;
    private Boolean isPassed;
    private Date startedAt;
    private Date submittedAt;
    private Integer timeSpent;
    private List<QuizAttemptAnswerResponse> answers;
}

File: react-type-vite/src/services/api/response/quizAttemptResponse.ts
import type { QuizAttemptAnswerResponse } from "./quizAttemptAnswerResponse"

export interface QuizAttemptResponse {
  id: number
  userId: string
  quizId: number
  score: number
  totalScore: number
  isPassed: boolean
  startedAt: Date
  submittedAt: Date
  timeSpent: number
  answers: QuizAttemptAnswerResponse[]
}